extensions/ql-vscode/src/view/variant-analysis/__tests__/VariantAnalysisStatusStats.spec.tsx

```diff
@@ -0,0 +1,74 @@
+import * as React from "react";
+import { render as reactRender, screen } from "@testing-library/react";
+import { VariantAnalysisStatus } from "../../../remote-queries/shared/variant-analysis";
+import {
+  VariantAnalysisStatusStats,
+  VariantAnalysisStatusStatsProps,
+} from "../VariantAnalysisStatusStats";
+import { formatDate } from "../../../pure/date";
+
+describe(VariantAnalysisStatusStats.name, () => {
+  const onViewLogsClick = jest.fn();
+
+  afterEach(() => {
+    onViewLogsClick.mockReset();
+  });
+
+  const render = (props: Partial<VariantAnalysisStatusStatsProps> = {}) =>
+    reactRender(
+      <VariantAnalysisStatusStats
+        variantAnalysisStatus={VariantAnalysisStatus.InProgress}
+        {...props}
+      />,
+    );
+
+  it("renders an in-progress status correctly", () => {
+    const { container } = render({
+      variantAnalysisStatus: VariantAnalysisStatus.InProgress,
+    });
+
+    expect(
+      container.getElementsByClassName(
+        "codicon codicon-loading codicon-modifier-spin",
+      ).length,
+    ).toEqual(1);
```

Comment:
I think `toBeInTheDocument` is sufficient for this, I don't think we need to explicitly test that we only have 1 element.

Should we also make the `Icon` more accessible by adding some alt text so we can use `screen.getByRole` instead? I'm not sure that should be part of this PR though.

Reply:
Definitely can do. I still need to access the first element of the array, but that works.

I don't have strong preferences about the icon. It was a bit annoying having to use this method to find the element, and it could be a bit fragile if we change the classes on the spinner. But also adding user-visible text to support the tests sounds a bit off too, perhaps. I think I'll leave it be for this PR at least.

Reply:
Definitely, I don't think we'd need to add user-visible text, but more something along the lines of [this](https://github.com/github/vscode-codeql/blob/27b4203d8ba85ef371f17aed0d167010119df679/extensions/ql-vscode/src/view/common/icon/Codicon.tsx#L17-L21) where we're adding an `aria-label` so screen readers understand it. We could potentially even use the `Codicon` component, although I'm not sure why that hadn't been done originally here. I suspect it's because of the `font-size: 1em`, but we'd need to test that theory.